### PR TITLE
Add vsthrottle.remove_bucket()

### PR DESCRIPTION
**Pull request by sumit-kushwah**
VOID remove_bucket(STRING key, INT limit, DURATION period, DURATION block)

Arguments:
  - Same arguments as is_denied()

Description

  Remove the token bucket identified by the four parameters, if it exists.

Example:

     sub vcl_deliver {
	    vsthrottle.remove_bucket(client.identity, 15, 10s, 30s);
     }

it fixes the issue #204

### CODE REVIEW

File: src/vmod_vsthrottle.c
Comment (gquintard) [inline src/vmod_vsthrottle.c:332]: indentation

File: src/vmod_vsthrottle.c
Comment (gquintard) [inline src/vmod_vsthrottle.c:318]: need to make sure `key` isn't NULL

Comment (gquintard) [inline src/vmod_vsthrottle.c:318]: you should be able to use to either use `get_bucket`, or at the very least point `do_digest` at `&k.digest` and save yourself a `memcpy`, no?

**Review by gquintard — CHANGES_REQUESTED** [review]
thank you for opening that PR!

I have a few changes I'd like to see before this can be merged, as well as some tests, of course :-)
Apart from that, I don't think that the new feature is contentious, but I'll have to check with the other maintainers to push it through.

Let me know if you have any questions!

**Comment by sumit-kushwah** [comment]
Ok, @gquintard Thanks for reviewing, I will make the changes as you suggested and write test cases.

**Comment by sumit-kushwah** [comment]
Hi, @gquintard  I have made the changes you recommended and wrote test cases for the new `remove_bucket` function. Please let me know if there are any remaining issues that need to be addressed, or if there are any other areas of concern.

File: src/vmod_vsthrottle.c
Comment (gquintard) [inline src/vmod_vsthrottle.c:326]: nitpick: you can just do
```
b = get_bucket(digest, limit, period, VTIM_mono());
```

and save a couple of lines by not having the `now` variable

**Review by gquintard — CHANGES_REQUESTED** [review]

File: src/vmod_vsthrottle.c
Comment (sumit-kushwah) [reply src/vmod_vsthrottle.c]: Thanks, done!

**Comment by gquintard** [comment]
great, looks good on my side, I'll pull in the others to see what they have to say.

hl @nigoroll, @hermunn, @Dridi

**Comment by sumit-kushwah** [comment]
@daghf, Done.

**Comment by daghf** [comment]
Thanks a lot for the contribution, @sumit-kushwah

**Comment by sumit-kushwah** [comment]
@gquintard, @daghf 
Thank you for taking the time to review and merge my pull request. I really appreciate it!

**Pull request merged**
